react/features/conference/middleware.js

```diff
@@ -48,3 +51,25 @@ MiddlewareRegistry.register(store => next => action => {
 
     return result;
 });
+
+/**
+ * Set up state change listener to perform maintenance tasks when the conference
+ * is left or failed, close all dialogs and unpin any pinned participants.
+ */
+StateListenerRegistry.register(
+    state => getCurrentConference(state),
+    (conference, { dispatch, getState }, prevConference) => {
+        const { authRequired, passwordRequired }
+            = getState()['features/base/conference'];
+
+        if (conference !== prevConference) {
+            dispatch(pinParticipant(null));
+
+            if (typeof authRequired === 'undefined'
+                    && typeof passwordRequired === 'undefined') {
+                // Conference changed, left or failed... and there is no
+                // pending authentication, close any dialog we might have open.
+                dispatch(hideDialog());
```

Comment:
What is the interaction like here with the web post-meeting feedback modal? It gets displayed okay?

Reply:
Good question. I'll test it out now.

Reply:
Well, fuck me. It's broken :-/

Reply:
Oh, I'm sorry to hear. Although, please note that the same dialog will appear on mobile soon too

Reply:
An interesting case here is that from the SDK events perspective something will have to change, because once the conference is over and the user is told about it, it will hide the view...

Reply:
Ok, I have to take a look how I implemented it (it's in a local copy here on my computer) so feel free to ignore this concern for now.

Reply:
I can definitely see there being a lot of stickiness around this feedback dialog, because on web the feedback dialog exists after conference exit and there really isn't much in the way of cleanup. Plus there's the wrinkle of not wanting to show the feedback dialog (on web at least) if feedback has already been submitted, and that submission state is stored in redux and not the conference itself. 

Maybe instead of hiding the feedback dialog explicitly show it? That might clash with any password entry though.